src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs

```diff
@@ -0,0 +1,666 @@
+        }
+
+        [Fact]
+        public void UsingVariableCannotBeReAssigned()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        using var x = (IDisposable)null;
+        x = null;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,9): error CS1656: Cannot assign to 'x' because it is a 'using variable'
+                //         x = null;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "x").WithArguments("x", "using variable").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void UsingVariableCannotBeUsedAsOutVariable()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Consume(out IDisposable x) 
+    {
+        x = null;
+    }
+
+    static void Main()
+    {
+        using var x = (IDisposable)null;
+        Consume(out x);
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (13,21): error CS1657: Cannot use 'x' as a ref or out value because it is a 'using variable'
+                //         Consume(out x);
+                Diagnostic(ErrorCode.ERR_RefReadonlyLocalCause, "x").WithArguments("x", "using variable").WithLocation(13, 21)
+                );
+        }
+
+        [Fact]
+        public void UsingVariableMustHaveInitializer()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        using IDisposable x;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (7,27): error CS0210: You must provide an initializer in a fixed or using statement declaration
+                //         using IDisposable x;
+                Diagnostic(ErrorCode.ERR_FixedMustInit, "x").WithLocation(7, 27)
+                );
+        }
+
+        [Fact]
+        public void UsingVariableFromExistingVariable()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var x = (IDisposable)null;
+        using var x2 = x;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void UsingVariableFromExpression()
+        {
+            var source = @"
+using System;
+class C
+{
+    static IDisposable GetDisposable()
+    {
+        return null;
+    }
+
+    static void Main()
+    {
+        using IDisposable x = GetDisposable();
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void UsingVariableFromAwaitExpression()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+class C
+{
+    static Task<IDisposable> GetDisposable()
+    {
+        return Task.FromResult<IDisposable>(null);
+    }
+
+    static async Task Main()
+    {
+        using IDisposable x = await GetDisposable();
+    }
```

Comment:
>} [](start = 4, length = 1)

Are we testing execution of such a method, to ensure `Dispose` is called if and only if the `await` completes successfully?